llvm/CMakeLists.txt

```diff
@@ -275,30 +275,47 @@ set(LLVM_CCACHE_BUILD OFF CACHE BOOL "Set to ON for a ccache enabled build")
 if(LLVM_CCACHE_BUILD)
   find_program(CCACHE_PROGRAM ccache)
   if(CCACHE_PROGRAM)
+    # ccache --version example output: "ccache version 4.9.1\n(..)"
+    execute_process(COMMAND ${CCACHE_PROGRAM} --version OUTPUT_VARIABLE CCACHE_VERSION_STR)
+    # Strip the prefix before the version number, taking some extra characters
+    string(SUBSTRING "${CCACHE_VERSION_STR}" 15 10 CCACHE_VERSION_STR)
```

Comment:
Since the version has a fixed format, could we use regular expression instead?

Reply:
Ah indeed that's a much better way to do it, thanks!